src/Sylius/Bundle/ApiBundle/ApiPlatform/Routing/IriConverter.php

```diff
@@ -28,22 +29,23 @@ public function __construct(
         private IriConverterInterface $decoratedIriConverter,
         private PathPrefixProviderInterface $pathPrefixProvider,
         private OperationResolverInterface $operationResolver,
+        private RouterInterface $router,
     ) {
     }
 
-    public function getResourceFromIri(string $iri, array $context = [], Operation $operation = null): object
+    public function getResourceFromIri(string $iri, array $context = [], ?Operation $operation = null): object
     {
         return $this->decoratedIriConverter->getResourceFromIri($iri, $context, $operation);
     }
 
     public function getIriFromResource(
         object|string $resource,
         int $referenceType = UrlGeneratorInterface::ABS_PATH,
-        Operation $operation = null,
+        ?Operation $operation = null,
         array $context = [],
     ): ?string {
         $resourceClass = $context['force_resource_class'] ?? (\is_string($resource) ? $resource : $this->getObjectClass($resource));
-        $pathPrefix = $this->pathPrefixProvider->getPathPrefix($context['request_uri'] ?? '');
+        $pathPrefix = $this->pathPrefixProvider->getPathPrefix($context['request_uri'] ?? $this->router->getContext()->getPathInfo());
```

Comment:
I've got few cases (for shop endpoint) where in the response iri was from admin section. After debugging I've added this, because sometimes `$context` was an empty array